src/librustc_driver/driver.rs

```diff
@@ -926,6 +926,7 @@ pub fn phase_3_run_analysis_passes<'tcx, F, R>(sess: &'tcx Session,
             passes.push_pass(
                 box mir::transform::simplify_branches::SimplifyBranches::new("initial"));
             passes.push_pass(box mir::transform::simplify_cfg::SimplifyCfg::new("qualify-consts"));
+            passes.push_pass(box mir::transform::copy_prop::CopyPropagation);
```

Comment:
Should go to another set of passes in stage 4 of the pipeline (down below in the source).